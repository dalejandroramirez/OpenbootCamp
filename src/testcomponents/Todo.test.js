import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Todo from './Todo';

// Borrar los renderizados anteriores
afterAll(() => cleanup());


describe('Todo component is working correctly', () => {
  const todo = {id: 5, text: 'Hacer las compras', completed: true}
  render(<Todo todo={todo} />);
  
  const todoelement = screen.getByTestId(`todo-test-${todo.id}`);
  const checkbox = todoelement.querySelector('#checkbox');

  it ('Should render Todo Component', () => {
    expect(todoelement).toBeInTheDocument();
  });

  it('Should have rigth text', () => {
    expect(todoelement).toHaveTextContent(todo.text);
  });
});

describe('Todo component should check/uncleck completed/not completed',() => {
  const todo2 = {id: 6, text: 'Hacer las compras', completed: true}
  render(<Todo todo={todo2} />);
  
  const todoelement = screen.getByTestId(`todo-test-${todo2.id}`);
  const checkbox2 = todoelement.querySelector('#checkbox');
 
  it('Should be checked', () => {
    expect(checkbox2).toBeChecked();
  })

  it('matches snapshot', () => {
    const snapTodo = {id: 1, text: 'Acabar esto nuevamente', completed: true}
    const tree = renderer.create(<Todo todo={snapTodo}/>)
    expect(tree).toMatchSnapshot();
  })
})

