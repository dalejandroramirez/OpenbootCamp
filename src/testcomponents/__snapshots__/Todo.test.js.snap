// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should check/uncleck completed/not completed matches snapshot 1`] = `
<div
  data-testid="todo-test-1"
>
  <label>
    <input
      checked={true}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    Acabar esto nuevamente
  </label>
</div>
`;
